src/dynamo-streams.test.ts

```diff
@@ -508,4 +514,227 @@ describe('DynamoStreamHandler', () => {
+                NewImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-1',
+                }) as any,
+              },
+            },
+            {
+              eventName: 'MODIFY',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-1', name: 'test-name-1' }) as any,
+                NewImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-1',
+                }) as any,
+                OldImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-2',
+                }) as any,
+              },
+            },
+            {
+              eventName: 'MODIFY',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-2', name: 'test-name-1' }) as any,
+                NewImage: marshall({
+                  id: 'test-id-2',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-1',
+                }) as any,
+                OldImage: marshall({
+                  id: 'test-id-2',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-2',
+                }) as any,
+              },
+            },
+            {
+              eventName: 'REMOVE',
+              dynamodb: {
+                Keys: marshall({ id: 'test-id-1', name: 'test-name-1' }) as any,
+                OldImage: marshall({
+                  id: 'test-id-1',
+                  name: 'test-name-1',
+                  otherValue: 'test-value-2',
+                }) as any,
+              },
+            },
+          ],
+        },
+        {} as any,
+        null as any,
+      );
+
+      const end = Date.now();
+
+      // This assertion confirms that the group doesn't process in less than 200ms.
+      // If it did, then some of the ordered events would be parallelized, which would be bad.
+      expect(end - start).toBeGreaterThanOrEqual(300);
+
+      // This assertions confirms that there is some parallelization happening.
+      expect(end - start).toBeLessThan(400);
+
+      // Now, let's also explicitly assert that the events were processed in order.
+      const insertFinishedTime = mocks.insert.finished.mock.calls.find(
+        ([, { id }]) => id === 'test-id-1',
+      )[0];
+      const modifyStartedTime = mocks.modify.started.mock.calls.find(
+        ([, { id }]) => id === 'test-id-1',
+      )[0];
+
+      expect(modifyStartedTime).toBeGreaterThanOrEqual(insertFinishedTime);
+
+      const modifyFinishedTime = mocks.modify.finished.mock.calls.find(
+        ([, { id }]) => id === 'test-id-1',
+      )[0];
+
+      const removeStartedTime = mocks.remove.started.mock.calls.find(
+        ([, { id }]) => id === 'test-id-1',
+      )[0];
+
+      expect(removeStartedTime).toBeGreaterThanOrEqual(modifyFinishedTime);
```

Comment:
To make sure things are really happen serially, should we modify this test case to exclusively check for `toBeGreaterThan`? Instead of `toBeGreaterThanOrEqual`?

Reply:
Good question. While you're conceptually right, it's possible that a full millisecond does not pass between the _end_ of one event and the _start_ of another. I originally used `toBeGreaterThan`, but I encountered failures even when just running locally.

Reply:
Gotcha. I wasn't going to mention this, but you saying that made me think about it again. Should we move to using fake timers to advance time instead of `wait`ing?

Reply:
I'd like to avoid it for now -- it might be better practice, but I think the tests are solid as-is @aecorredor